eng/scripts/TypeSpec-Generate-Sdk.ps1

```diff
@@ -0,0 +1,35 @@
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $SdkRepoRootDirectory,
+  [Parameter(Position = 1)]
+  [string] $TypeSpecProjectDirectory = ".", # A directory of `tspconfig.yaml` or a remoteUrl of `tspconfig.yaml`
```

Comment:
I think the usage between this and commithash seem to conflict.  If I use a local directory there won't be a commithash right?  If I use a uri then their will be but perhaps we can get that for them?

I wonder if we can collapse `TypeSpecProjectDirectory`, `CommitHash`, and `RepoUrl` into 1 param `TspConfigLocation`.  If its a url, we know the repoUrl and we can query for the latestHash if its not in the url.  If its a local file then we don't need either of those things.

Reply:
@m-nash , 
Both `CommitHash` and `RepoUrl` are required to create `tsp-location.yaml`, but script would compute the value if they're not provided in input. 

We already support what you said here as below, let me know if you think there're gaps.

- if `$TypeSpecProjectDirectory` is a url of tspconfig.yaml, it would compute the commit from url then use that to fetch from remote repo.
- if `$TypeSpecProjectDirectory` is a local folder of tspconfig.yaml, it would use the local spec repo HEAD commit and repo info to create `tsp-location.yaml`. Then it would use the local specs to generate SDK instead of fetching the commit from remote.
- if spec writer explicitly pass in `CommitHash` and `RepoUrl`, then they would be respected in `tsp-location.yaml` and script would fetch the commit from remote spec repo to generate sdk.

Reply:
gotcha if we don't want to collapse the params then perhaps some documentation that these are meant to be used in only mutually exclusive manner.  Its not clear which ones I should pass and when.